cpu/rpx0xx/periph/adc.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2023 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_rpx0xx
+ * @ingroup     drivers_periph_adc
+ * @{
+ *
+ * @file
+ * @brief       Low-level ADC driver implementation
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ * @}
+ */
+
+#include <errno.h>
+#include "cpu.h"
+#include "mutex.h"
+#include "periph/adc.h"
+#include "periph_conf.h"
+
+#define _PADS_BANK0_PIN_REG(x) ((uint32_t*)(((PADS_BANK0_BASE + (x * 4)+4)) ))
+
+/**
+ * @brief   Lock to prevent concurrency issues when used from different threads
+ */
+static mutex_t lock;
+
+static inline void prep(void)
+{
+    mutex_lock(&lock);
+}
+
+static inline void done(void)
+{
+    mutex_unlock(&lock);
+}
+
+static void _disable_digital_func(gpio_t pin) {
+    /* OD et IE mask bit are the same for all pins */
+    volatile uint32_t* reg = _PADS_BANK0_PIN_REG(pin);
+    /* Set output disable bit */
+    io_reg_atomic_set(reg, 1 << PADS_BANK0_GPIO0_OD_Pos);
+    /* Unset output enable */
+    io_reg_atomic_clear(reg, 1 << PADS_BANK0_GPIO0_IE_Pos);
```

Comment:
I should have used `gpio_set_pad_config()` instead of this dirty hack.
I'll fix it tomorrow.